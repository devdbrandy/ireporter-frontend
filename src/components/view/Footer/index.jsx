import React from 'react';
import PropType from 'prop-types';

const Footer = ({ customClass, contentClass }) => (
  <footer className={customClass || 'main--footer'}>
    <div className={contentClass || 'footer--content'}>
      <p className="credit">
        Made with
        {' '}
        <span>
          <i className="fas fa-heart" />
        </span>
        {' '}
        by
        {' '}
        <a href="https://github.com/devdbrandy">Dbrandy</a>
      </p>
      <p>
        Copyright &copy;
        {' '}
        <span id="copyright-year">2018</span>
        {' '}
        iReporter. All Rights Reserved.
      </p>
    </div>
  </footer>
);

Footer.propTypes = {
  customClass: PropType.string,
  contentClass: PropType.string,
};

Footer.defaultProps = {
  customClass: '',
  contentClass: '',
};

export default Footer;
