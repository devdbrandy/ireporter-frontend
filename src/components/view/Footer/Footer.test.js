import React from 'react';
import { shallow } from 'enzyme';
import Footer from './index';

describe('<Footer />', () => {
  it('renders without crashing given the required props', () => {
    const enzymeWrapper = shallow(<Footer />);
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
