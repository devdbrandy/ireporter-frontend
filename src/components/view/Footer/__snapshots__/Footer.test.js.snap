// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders without crashing given the required props 1`] = `
<footer
  className="main--footer"
>
  <div
    className="footer--content"
  >
    <p
      className="credit"
    >
      Made with
       
      <span>
        <i
          className="fas fa-heart"
        />
      </span>
       
      by
       
      <a
        href="https://github.com/devdbrandy"
      >
        Dbrandy
      </a>
    </p>
    <p>
      Copyright Â©
       
      <span
        id="copyright-year"
      >
        2018
      </span>
       
      iReporter. All Rights Reserved.
    </p>
  </div>
</footer>
`;
